### PR TITLE
Header Accessibility

### PR DESCRIPTION
**Pull request by rebeccacremona**
- Hover/focus styles for the logo link
- "Create casebook" button is a real button
- The modal that it opens has all the behavior described here: https://github.com/harvard-lil/accessibility-tools/tree/master/code/modals
- The buttons/links are all the same size
- User picture removed from display per team decision!
- Dropdown coded up as a touch- and keyboard-usable Bootstrap dropdown
- Dropdown made white and full-width per Adam.

image
(The screenshot removed my cursor, which was hovering over "Edit", which is why that's gray. keyboard focus is on the button, which is why that has focus styles. )

Lemme know if you don't like anything about the way this looks!

Not in this PR:
- The modal accessibility fixes will need to be applied to every modal. So, I should abstract that out into a more generic component. If you have no objections, I'd like to do that when fixing the next modal, rather than right now. (Having two examples will make it easier to get the abstraction right, I think.)
- If the user picture is not being displayed, then the ability to upload one can probably be removed from the code base, but that feels like a separate project.
- The generic focus styles should, ideally, be applied to everything, rather than being applied to a few elements at a time, like I'm doing here. But, when I tried that, weird visual relics popped up in unexpected places. I'll fix that as I go along, and then hopefully can switch over to a general declaration.

### CODE REVIEW

**Comment by cgruppioni** [comment]
I just created issue #354 to address `If the user picture is not being displayed, then the ability to upload one can probably be removed from the code base, but that feels like a separate project.`

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -7,7 +7,7 @@
     </div>
     <div class="links">
       <% if current_user.present? %>
-        <div class="create-casebook" data-action="show-casebook-modal"><%= t('header.links.create-casebook') %></div>
+        <button type="button" class="create-casebook" data-action="show-casebook-modal"><%= t('header.links.create-casebook') %></button>
```
Comment (cgruppioni) [inline app/views/layouts/_header.html.erb]: Is there a reason you used `button` instead of `button_to`?

Comment (rebeccacremona) [reply app/views/layouts/_header.html.erb]: Aha. Sorry, I'm not used to Rails' view helpers. Can you show me how to use `button_to` here?

Comment (rebeccacremona) [reply app/views/layouts/_header.html.erb]: Nevermind, I should be able to figure it out. Fix forthcoming.

Comment (rebeccacremona) [reply app/views/layouts/_header.html.erb]: Aha! Yes, there is a reason.  If this is the right `button_to` (https://apidock.com/rails/ActionView/Helpers/UrlHelper/button_to), it sounds ike `button_to` creates a whole form that, when you click on the button, submits the form. Here, we just need a button to trigger some javascript that opens the modal and performs related housekeeping: no HTTP request involved.

Comment (cgruppioni) [reply app/views/layouts/_header.html.erb]: Ah! Ok --- I was wrong.

**Pull request merged**